### PR TITLE
My heap

### PR DESCRIPTION
**Pull request by DanielLavie**

### CODE REVIEW

File: src/MyBinaryTree.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+                    nextLevel.add(currentNode.getLeft());
+                }
+                if (currentNode.getRight() != null) {
+                    nextLevel.add(currentNode.getRight());
+                }
+                System.out.print(currentNode.getData().getAge() + " ");
+            }
+            System.out.println();
+            currentLevel = nextLevel;
+            nextLevel = new LinkedList<TreeNode>();
```
Comment (xenuvix) [inline src/MyBinaryTree.java]: Replace this with MyLinkedList,

File: src/MyBinaryTree.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+            leftChild = currentNode.getLeft();
+            rightChild = currentNode.getRight();
+        }
+    }
+
+    /**
+     * Purely for testing purpose!
+     */
+    public void printTree() {
+        Queue<TreeNode> currentLevel = new LinkedList<TreeNode>();
```
Comment (xenuvix) [inline src/MyBinaryTree.java]: Don't think you can use Queue here. Am I wrong?

File: src/MyBinaryTree.java [hunk 3]
```diff
@@ -0,0 +1,269 @@
+import java.util.Iterator;
+import java.util.LinkedList;
```
Comment (xenuvix) [inline src/MyBinaryTree.java]: Remove after replacing with MyLinkedList

File: src/MyBinaryTree.java [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+/**
+ * This class represents a binary tree with additional features specifically
+ * for a binary heap.
+ * The tree is indexed based on breadth-first order.
+ * Note that this tree is only capable of holding Person data objects
+ */
+public class MyBinaryTree {
+
+    TreeNode root;
```
Comment (xenuvix) [inline src/MyBinaryTree.java]: Set those to be private

File: src/MyBinaryTree.java [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+     *
+     * @param binaryTreeArrayRepresentation Array representation of a binary heap.
+     */
+    public MyBinaryTree(Person[] binaryTreeArrayRepresentation) {
+
+        // Create a temporary array to allow the construction of the tree
+        // Java will initiate each of the object to null by default
+        TreeNode[] nodesTreeOrder = new TreeNode[binaryTreeArrayRepresentation.length + 1];
+
+        // Create the the nodes inside the temp array, O(N).
```
Comment (xenuvix) [inline src/MyBinaryTree.java]: the the? :)

File: src/MyBinaryTree.java [hunk 6]
```diff
@@ -0,0 +1,269 @@
+import java.util.Iterator;
```
Comment (xenuvix) [inline src/MyBinaryTree.java]: Why did you add an src folder? The automatic tester tries to look for files inside the root folder given.

File: src/MyHeap.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+        int leftChildIndex = getLeftChildIndex(persons, index);
+        int rightChildIndex = getRightChildIndex(persons, index);
+
+        // Check if one of the childes is older than the index
+
+        int oldestPerson = index;
+
+        if (indexInsideArray(persons, leftChildIndex)) {
+            if (persons[leftChildIndex].getAge() >=
```
Comment (xenuvix) [inline src/MyHeap.java]: If this is ==, do you really need to set oldestPerson = leftChildIndex?
Same for next if.
Let's try to think what happens it it is equal.

File: src/MyBinaryTree.java (hunk 6 above)
Comment (DanielLavie) [inline src/MyBinaryTree.java]: I organized it as src and test folder for better seperation, didn't think about it. ill fix it.

File: src/MyBinaryTree.java (hunk 2 above)
Comment (DanielLavie) [inline src/MyBinaryTree.java]: Actually this is a method for testing purpose only (as shown in the comment above). You think they will allow the use of java DS here or should i remove the method?

File: src/MyBinaryTree.java (hunk 1 above)
Comment (DanielLavie) [inline src/MyBinaryTree.java]: See previous reply

File: src/MyBinaryTree.java (hunk 2 above)
Comment (xenuvix) [inline src/MyBinaryTree.java]: I’m not sure. I guess you can just add a comment explaining.
